django-rgd-imagery/rgd_imagery/rest/viewsets.py

```diff
@@ -76,11 +76,22 @@ class RasterViewSet(ModelViewSet, TaskEventViewSetMixin):
     filterset_class = filters.RasterMetaFilter
 
     def get_serializer_class(self):
-        if self.action in {'list', 'retrieve'}:
+        if self.action in {'list', 'retrieve', 'status'}:
             return serializers.RasterMetaSerializer
+        if self.action in {'stac'}:
+            return stac.serializers.ItemSerializer
         return serializers.RasterSerializer
 
     def get_queryset(self):
-        if self.action in {'list', 'retrieve'}:
+        if self.action in {'list', 'retrieve', 'stac', 'status'}:
             return models.RasterMeta.objects.all()
         return models.Raster.objects.all()
+
+    @swagger_auto_schema(
+        method='GET',
+        operation_summary='Fetch the STAC Item JSON for this raster.',
+    )
+    @action(detail=True)
+    def stac(self, *args, **kwargs):
+        obj = self.get_object()
+        return Response(stac.serializers.ItemSerializer().to_representation(obj))
```

Comment:
@mcovalt, is there a better way to do this, since `stac.serializers.ItemSerializer` is handled in `get_serializer_class()`?

Reply:
You can override the `queryset` and `serializer_class` by passing them in as kwargs to the action decorator. https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing